victims/victims.go

```diff
@@ -257,3 +241,78 @@ func RandomPodName(pods []v1.Pod) string {
 	randIndex := r.Intn(len(pods))
 	return pods[randIndex].Name
 }
+
+// Returns the number of pods to kill based on the number of all running pods
+func (v *VictimBase) KillNumberForKillingAll(clientset kube.Interface, killPercentage int) int {
+	pods, err := v.RunningPods(clientset)
+	if err != nil {
+		glog.V(6).Infof("Failed to get list of running pods %s %s", killPercentage, v.kind, v.name)
+		return 0
+	}
+
+	killNum := len(pods)
+
+	return killNum
+}
+
+// Returns the number of pods to kill based on a kill percentage and the number of running pods
+func (v *VictimBase) KillNumberForFixedPercentage(clientset kube.Interface, killPercentage int) int {
+	if killPercentage == 0 {
+		glog.V(6).Infof("Not terminating any pods for %s %s as kill percentage is 0\n", v.kind, v.name)
+		// Report success
+		return 0
+	}
+	if killPercentage < 0 {
+		glog.V(6).Infof("Expected kill percentage config %d to be between 0 and 100 for %s %s. Defaulting to 0", killPercentage, v.kind, v.name)
+		killPercentage = 0
+	}
+	if killPercentage > 100 {
+		glog.V(6).Infof("Expected kill percentage config %d to be between 0 and 100 for %s %s. Defaulting to 100", killPercentage, v.kind, v.name)
+		killPercentage = 100
+	}
+
+	pods, err := v.RunningPods(clientset)
```

Comment:
minor comment but the code to get list of all running pods seems to be repeated thrice. Can we move this to a function.

Reply:
Pr updated.